File: cpp_inheritance/Critter.cpp
#include <string>
#include <iostream>

#include "Critter.h"

//using namespace std;

//Constructor and greet are defined for Critter
Critter::Critter(){
  Critter::name = "anonymous";
} // end constructor

Critter::Critter(std::string name){
  Critter::name = name;
}

void Critter::greet(){
  std::cout << "My name is " << Critter::name << std::endl;
} // end greet

/*
int main(){

  std::cout << "main class: ";
  Critter c;
  c.greet();

  return 0;
}
*/

File: java_jar/Critter.java

public class Critter {
	protected String name;
	
	public static void main(String[] args) {
		//test harness
		
		//test null param constructor
		Critter c = new Critter();
		
		//test methods
		c.sayHi();
		c.setName("George");
		c.sayHi();
		System.out.println(c.getName());
		
		//test single param constructor
		c = new Critter("Martha");
		c.sayHi();
		
		/*  EXPECTED OUTPUT:
		 * Hi, my name is anonymous
		 * Hi, my name is George
		 * George
		 * Hi, my name is Martha
		 */
		
	} // end main
	
	public Critter(){
		this.name = "anonymous";
	} // end constructor
	
	public Critter(String name){
		this.name = name;
	} // end constructor
	
	public void sayHi(){
		System.out.println("Hi, My name is " + this.name);
	} // end sayHi
	
	public void setName(String name){
		this.name = name;
	} // end setName
	
	public String getName(){
		return this.name;
	} // end getName
} // end class def
